### ISSUE TITLE
DataTable - add support for selecting checkboxes via 'enter' key

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#7788]
Currently we only support 'space' key for interacting with the checkboxes in a DataTable via keyboard. We should add support for 'space' as well

**Comment by arsh342** [comment]
can you please explain more

**Comment by NOMADE55** [comment]
I was interested in doing this for Hacktoberfest, but it seems this is already done 😅. 

https://github.com/grommet/grommet/blob/cf0b1791da6eaaa439d43a535ae110271d028d08/src/js/components/DataTable/Body.js#L303-L335

**Comment by britt6612** [comment]
@NOMADE55

Hmm, I’m not entirely sure, I haven’t looked too deeply into the code yet.

You can see the behavior in Storybook here:
https://storybook.grommet.io/?path=/story/visualizations-datatable-controlled--controlled-data-table&globals=theme:hpe

If you navigate to a checkbox within the DataTable using the keyboard and press `Enter`, it won’t get selected. However, pressing `Space` will correctly check it.

**Comment by NOMADE55** [comment]
Ah I see, this behavior seems linked to the normal browser keyboard navigation, which uses "space" for selecting the checkbox.
The current Keyboard implementation expects the row to be clickable.

**Comment by MikeKingdom** [comment]
Note that in general checkboxes should only be selectable via the spacebar and not by Enter.

**Comment by rickyentzer** [comment]
I reviewed checkbox selection on [Webaim](https://webaim.org/techniques/keyboard/) and [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/checkbox) and also looked a several other design systems ([IBM](https://carbondesignsystem.com/components/data-table/usage/), [Material](https://m2.material.io/components/data-tables), and https://baseweb.design/components/data-table/) to see how they handled data-tables with checkboxes in each row. I also reached out to Bill Tipton for his guidance on the subject.

The consensus and pattern is that when using a keyboard, checkbox selection should be triggered only by the spacebar. It is also the default behavior in browsers.

**Comment by jcfilben** [comment]
Thanks for researching! I'm going to go ahead and close this issue and mark it as not planned
